src/WholeBodyStateDisplay.cpp

```diff
@@ -679,6 +678,13 @@ void WholeBodyStateDisplay::processWholeBodyState() {
     Eigen::Vector3d for_ref_dir = -Eigen::Vector3d::UnitZ();
     Eigen::Vector3d for_dir(contact.wrench.force.x, contact.wrench.force.y, contact.wrench.force.z);
 
+    // Getting the contact's center of pressure
+    // NOTE: x component is negative due to right-hand rotation rule
+    Eigen::Vector3d cop_pos = Eigen::Vector3d(-contact.wrench.torque.y / contact.wrench.force.z,
+                                              contact.wrench.torque.x / contact.wrench.force.z,
+                                              0.0);  // Origin of frame is already at contact position
+    Ogre::Vector3 cop_point(cop_pos(0), cop_pos(1), cop_pos(2));
+
```

Comment:
The key change is moving this CoP calculation from the CoP-specific block to the for loop scope.